### ISSUE TITLE
ANR issues due to `getSubNodes` recursion depth

### ISSUE THREAD

**Issue by saurabhkumar8112** [X-SLAYER/flutter_accessibility_service#28]
When iterating through the child nodes of a parent node, the method `onAccessibilityEvent` invokes `getSubNodes`, a recursive function. This recursion, influenced by the depth of the tree (`treeDepth`), has the potential to cause a stack overflow, leading to Application Not Responding (ANR) errors.

There are [discussions](https://stackoverflow.com/questions/46484851/getting-lot-of-anr-using-accessibility-service-in-android) on StackOverflow, of ANRs occurring when the accessibility service is used for navigating through AccessibilityNodeInfo.

To mitigate this issue, I propose to add a new parameter, `maximumTreeDepth`, which would serve as a threshold for the recursion depth. Once this threshold is reached, further traversal operations would be halted, and the method `getNextTexts` can return.

This approach allows users to balance the trade-off between performance aspects (such as minimizing ANRs and reducing battery consumption) and usability, by adjusting the maximumTreeDepth parameter according to the needs of their specific application.

Let me know what do you think about it, I can then raise a PR for the same!

**Comment by X-SLAYER** [comment]
yes that's what i though about it will be better if you can raise an PR about this so we could test that

**Comment by X-SLAYER** [comment]
Check the latest [version](https://pub.dev/packages/flutter_accessibility_service). Now it is more optimized on the subNodes fetching
